pynobo.py

```diff
@@ -849,6 +849,31 @@ async def async_update_zone(self, zone_id, name=None, week_profile_id=None, temp
 
         await self.async_send_command(command)
 
+    def update_week_profile(self, week_profile_id, name=None, profile=None):
+        self._create_task(self.async_update_week_profile(week_profile_id, name, profile))
+
+    async def async_update_week_profile(self, week_profile_id, name=None, profile=None):
+        """
+        Update the name and profile parameter for a week.
+
+        :param week_profile_id: the week_profile_id
+        :param name: the new zone name (default None)
+        :param profile: the new profile (default None)
+        """
+
+        # Initialize command with the current zone settings
+        command = [nobo.API.UPDATE_WEEK_PROFILE] + list(self.week_profiles[week_profile_id].values())
+
+        # Replace command with arguments that are not None. Is there a more elegant way?
+        if name:
+            command[2] = name
+        if profile:
+            command[3] = profile
```

Comment:
Check that `profile` contains legal value(s)

Reply:
**Note:** From the API documentation:
> “Profile” can contain maximum 672 timestamps. Timestamps must have 00, 15, 30 or 45 for the minute field to be compatible with the app interface that only allows 15 minute intervals.

I'm not sure if we should enforce the minute field rule though.

Reply:
we should because we don't know what the App is doing and this could stop the App to work/ for normal user brick the Hub.

Reply:
Yes, after thinking on this I agree. This also applies for "from-to" overrides. It most likely won't brick the hub though, "only" break the app.